src/main/java/adris/altoclef/tasks/misc/GetToOuterEndIslandsTask.java

```diff
@@ -0,0 +1,120 @@
+    private Task _beatTheGame;
+    private boolean _pathingToGateway = false;
+    public final Vec3i[] OFFSETS = {
+            new Vec3i(1, -1, 1),
+            new Vec3i(1, -1, -1),
+            new Vec3i(-1, -1, 1),
+            new Vec3i(-1, -1, -1),
+            new Vec3i(2, -1, 0),
+            new Vec3i(0, -1, 2),
+            new Vec3i(-2, -1, 0),
+            new Vec3i(0, -1, -2)
+    };
+    public GetToOuterEndIslandsTask() {
+
+    }
+
+    @Override
+    protected void onStart(AltoClef mod) {
+        mod.getBehaviour().push();
+        mod.getBlockTracker().trackBlock(Blocks.END_GATEWAY);
+        _beatTheGame = new BeatMinecraft2Task();
+    }
+
+    @Override
+    protected Task onTick(AltoClef mod) {
+        if (isFinished(mod)) {
+            setDebugState("What?");
+            return null;
+        }
+        if (mod.getBlockTracker().anyFound(Blocks.END_GATEWAY)) {
+            if (!mod.getItemStorage().hasItemInventoryOnly(Items.ENDER_PEARL)) {
+                setDebugState("Getting an ender pearl");
+                return new CataloguedResourceTask(new ItemTarget(Items.ENDER_PEARL, 1));
+            }
+            BlockPos gateway = mod.getBlockTracker().getNearestTracking(Blocks.END_GATEWAY).get();
+            int blocksNeeded = Math.abs(mod.getPlayer().getBlockY() - gateway.getY()) +
+                    Math.abs(mod.getPlayer().getBlockX() - gateway.getX()) +
+                    Math.abs(mod.getPlayer().getBlockZ() - gateway.getZ()) - 1;
+            if (StorageHelper.getBuildingMaterialCount(mod) < blocksNeeded) {
+                setDebugState("Getting building materials");
+                return new GetBuildingMaterialsTask(blocksNeeded);
+            }
+            GoalAnd goal = makeGoal(gateway);
+            Debug.logMessage(mod.getPlayer().getBlockPos().toString());
+            if (!goal.isInGoal(mod.getPlayer().getBlockPos()) || !mod.getPlayer().isOnGround()) {
+                mod.getClientBaritone().getCustomGoalProcess().setGoal(goal);
+                if (!mod.getClientBaritone().getPathingBehavior().isPathing()) {
+                    mod.getClientBaritone().getCustomGoalProcess().path();
+                }
+                setDebugState("Getting close to gateway...");
+                return null;
+            }
+            setDebugState("Throwing the pearl inside");
+            return new InteractWithBlockTask(Items.ENDER_PEARL, gateway);
+        }
+        setDebugState("Beating the Game to get to an end gateway");
+        return _beatTheGame;
+    }
+
+    @Override
+    protected void onStop(AltoClef mod, Task interruptTask) {
+        mod.getBlockTracker().stopTracking(Blocks.END_GATEWAY);
+        mod.getBehaviour().pop();
+    }
+
+    @Override
+    protected boolean isEqual(Task other) {
+        return other instanceof GetToOuterEndIslandsTask;
+    }
+
+    @Override
+    public boolean isFinished(AltoClef mod) {
+        return WorldHelper.getCurrentDimension() == Dimension.END &&
```

Comment:
I would have a flag for if it successfully completed entering the portal as well. Are end islands always 400 meters away? Again, take this comment as a suggestion rather than change requisition.

Reply:
Actually, I'm not 100% sure how to do that in the case that it fails, like if we suddenly teleport? That's probably as reliable as this solution

Reply:
This seems fine. Minor changes, I'd use `WorldHelper.inRangeXZ` so it ignores your y position and a constant at the top of the file (ex. `private static final int OUTER_END_ISLAND_START_RADIUS = 400` or a shorter name if you want)

Reply:
Actually I did a bit of research and it starts around 1000 blocks away so I'll make it 800 just to be safe